doc/user/content/integrations/terraform.md

```diff
@@ -0,0 +1,224 @@
+Terraform plugin that allows Terraform to interact with the Materialize API.
+
+Materialize also maintains several modules that make it easier to manage your
+other cloud resources that Materialize depends on. Modules allow you to bypass
+manually configuring cloud resources and are an efficient way of deploying
+infrastructure with a single `terraform apply` command.
+
+## Terraform provider
+
+To use the Materialize provider, you create a new `main.tf` file and add the
+required providers:
+
+```hcl
+terraform {
+  required_providers {
+    materialize = {
+      source = "MaterializeInc/materialize"
+      version = "0.0.5"
+    }
+  }
+}
+```
+
+The Materialize provider is hosted on the [Terraform provider registry](https://registry.terraform.io/providers/MaterializeInc/materialize/latest).
+
+### Authentication
+
+To configure the provider to communicate with your Materialize organization, you
+need to authenticate with a Materialize username, app password, and other
+specifics from your account.
+
+Materialize recommends saving sensitive input variables as environment variables
+to avoid checking secrets into source control. To create a Terraform environment
+variable, export your sensitive variable in the terminal:
+
+```shell
+export TF_VAR_MZ_PW=yourMZpassword
+```
+
+In the `main.tf` file, add the provider configuration and any variable
+references:
+
+```hcl
+
+variable "MZ_PW" {}
+
+provider "materialize" {
+  host     = "yourMZhostname"
+  username = "yourMZusername"
+  password = var.MZ_PW
+  port     = 6875
+  database = "yourMZdatabase"
+}
+```
+
+### Create Materialize resources
+
+The Materialize provider allows you to create several resource types in your
+organization. Resources correspond to Materialize objects and are configured
+with the `resource` block in your Terraform configuration file.
+
+For example, to create a new cluster, you would use the `materialize_cluster`
+resource:
+
+```hcl
+resource "materialize_cluster" "example_cluster" {
+  name = "cluster"
+}
+```
+
+All the available Materialize resources and documentation can be found in the [Terraform provider
+registry](https://registry.terraform.io/providers/MaterializeInc/materialize/latest/docs).
+
+
+## Terraform modules
+
+Materialize objects can be created through standard SQL queries, but managing
+more than a few components can be difficult to maintain.  Using Terraform can be
+helpful when you manage other cloud resources in collaboration with Materialize.
+The Terraform modules below provide the cloud infrastructure foundation
+Materialize needs to communicate with components outside of Materialize itself.
+The Materialize provider allows users to manage Materialize resources in the
+same programmatic way.
+
+You can use the modules to establish the underlying cloud
+resources and then use the Materialize provider to build Materialize-specific
+objects. A few use cases are captured in the sections below:
```

Comment:
Same comment as above; we don't want to make it sound like these modules are officially supported by us. We've added a disclaimer to each module's repo stating just that:

> This is provided on a best-effort basis and Materialize cannot offer support for this module.

@jseldess, what do you think about just listing the modules with links to the repos and `/demos` to make this more understated?

Reply:
I think clarifying how much support users can expect is fair. Since they are in the demos repo, maybe changing the language in this doc would be more helpful. Something like 

> Materialize has several demonstration modules for some specific use cases. These modules are best-effort and should be used with caution.